sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_parse.py

```diff
@@ -0,0 +1,23 @@
+# -------------------------------------------------------------------------
```

Comment:
I think I understand what this comment
![image](https://github.com/user-attachments/assets/9cdf77d4-3900-407f-9334-3438f0c7858d)

means in your previous PR now 😄 

Was this something we discussed and landed on? I am asking because this super anti-pattern to what we have in all the other typing work (released or in-flight)
`_parse_url` lives in the `*****_helpers` of the relevant file and this is the case in Blobs, Queue, and in File-share. I don't think we should be making a brand new file to contain this parsing logic, especially since this would make Datalake the only one with this structure.

Reply:
As I mentioned in the other PR, if this function is actually the same for each client, it does make sense to have it not be duplicated in each client helper file and for it to live someone shared. I don't love creating a whole new file for it though.

If this function is the exact same for all clients **in all packages** if can move to `_shared.parsers`. If its specific to each package, find some other file to stick it in.

If we go the shared route, we should make a PR into `main` to adjust that now for all packages.